File: Java/TopKFrequentElements_347.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Scanner;

class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> dic = new HashMap<>();
        for (int num : nums) {
            dic.put(num, dic.getOrDefault(num, 0) + 1);
        }

        List<Map.Entry<Integer, Integer>> sorted_ls = new ArrayList<>(dic.entrySet());
        sorted_ls.sort((a, b) -> b.getValue() - a.getValue());

        int[] res = new int[k];
        for (int i = 0; i < k; i++) {
            res[i] = sorted_ls.get(i).getKey();
        }
        return res;
    }
}

public class TopKFrequentElements_347 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter number of elements: ");
        int n = sc.nextInt();

        int[] nums = new int[n];
        System.out.print("Enter the elements: ");
        for (int i = 0; i < n; i++) {
            nums[i] = sc.nextInt();
        }

        System.out.print("Enter k: ");
        int k = sc.nextInt();

        Solution sol = new Solution();
        int[] result = sol.topKFrequent(nums, k);

        System.out.println("Top " + k + " frequent elements:");
        for (int num : result) {
            System.out.print(num + " ");
        }
    }
}

File: LeetCode/Python/TopKFrequentElements_347.py
from typing import List
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dic={}
        for i in nums:
            if i in dic:
                dic[i]+=1
            else:
                dic[i]=1
        sorted_dic=sorted(dic.items(),key=lambda x :x[1],reverse=True)
        res=[i[0] for i in sorted_dic[:k] ]
        return res
    
s=Solution()

ls=list(map(int,input("Enter the elements : ").split()))
k=int(input('k : '))
print(f"Top {k} frequent elements : {s.topKFrequent(ls,k)}")
